modules/auxiliary/scanner/mssql/mssql_login.rb

```diff
@@ -35,14 +36,32 @@ def initialize
       OptBool.new('TDSENCRYPTION', [ true, 'Use TLS/SSL for TDS data "Force Encryption"', true]),
     ])
 
-    deregister_options('PASSWORD_SPRAY')
+    options_to_deregister = %w[PASSWORD_SPRAY]
+    if framework.features.enabled?(Msf::FeatureManager::MSSQL_SESSION_TYPE)
+      add_info('New in Metasploit 6.4 - The %grnCreateSession%clr option within this module can open an interactive session')
+    else
+      options_to_deregister << 'CreateSession'
+    end
+    deregister_options(*options_to_deregister)
+  end
+
+  def create_session?
+    if framework.features.enabled?(Msf::FeatureManager::MSSQL_SESSION_TYPE)
+      datastore['CreateSession']
+    else
+      false
+    end
   end
 
   def run_host(ip)
     print_status("#{rhost}:#{rport} - MSSQL - Starting authentication scanner.")
 
     if datastore['TDSENCRYPTION']
-      print_status("Manually enabled TLS/SSL to encrypt TDS payloads.")
+      if create_session?
+        raise "Cannot create sessions when encryption is enabled. See https://github.com/rapid7/metasploit-framework/issues/18745 to vote for this feature"
```

Comment:
```suggestion
        raise Msf::OptionValidateError.new(
          {
            'TDSENCRYPTION' => "Cannot create sessions when encryption is enabled. See https://github.com/rapid7/metasploit-framework/issues/18745 to vote for this feature"
          }
        )
```

### Before

Cryptic stacktrace:

```

msf6 auxiliary(scanner/mssql/mssql_login) > run 192.168.123.136  username=admin password=p4$$w0rd  tdsencryption=true createsession=true

[*] 192.168.123.136:1433  - 192.168.123.136:1433 - MSSQL - Starting authentication scanner.
[-] 192.168.123.136:1433  - Auxiliary failed: RuntimeError Cannot create sessions when encryption is enabled. See https://github.com/rapid7/metasploit-framework/issues/18745 to vote for this feature
[-] 192.168.123.136:1433  - Call stack:
[-] 192.168.123.136:1433  -   /Users/user/Documents/code/metasploit-framework/modules/auxiliary/scanner/mssql/mssql_login.rb:61:in `run_host'
[-] 192.168.123.136:1433  -   /Users/user/Documents/code/metasploit-framework/lib/msf/core/auxiliary/scanner.rb:128:in `block (2 levels) in run'
[-] 192.168.123.136:1433  -   /Users/user/Documents/code/metasploit-framework/lib/msf/core/thread_manager.rb:105:in `block in spawn'
[*] Auxiliary module execution completed
```

### After

Cleaner UI that aligns with other module validation

```
msf6 auxiliary(scanner/mssql/mssql_login) > reload
[*] Reloading module...
[*] New in Metasploit 6.4 - The CreateSession option within this module can open an interactive session
msf6 auxiliary(scanner/mssql/mssql_login) > run 192.168.123.136  username=admin password=p4$$w0rd  tdsencryption=true createsession=true

[*] 192.168.123.136:1433  - 192.168.123.136:1433 - MSSQL - Starting authentication scanner.
[-] 192.168.123.136:1433  - Msf::OptionValidateError The following options failed to validate:
[-] 192.168.123.136:1433  - Invalid option TDSENCRYPTION: Cannot create sessions when encryption is enabled. See https://github.com/rapid7/metasploit-framework/issues/18745 to vote for this feature
[*] Auxiliary module execution completed
msf6 auxiliary(scanner/mssql/mssql_login) > 
```